### PR TITLE
add post about laravel ecosystem

### PR DESCRIPTION
**Pull request by tonysm**
### Added

- [x] write a blog post about Laravel's Ecosystem;
- [x] add Cachier section;
- [x] add Socialite section;
- [x] re-write the queues section so it reads better;
- [x] re-write the conclusion;

### CODE REVIEW

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+date:   2015-05-08
+author: tony
+categories: ["laravel", "ecosystem", "development", "deploy", "fullstach", "framework"]
+excerpt: "Laravel is awesome. Not just the framework, but its ecosystem. Let me walk you through it and show some of the beauties arround this community. "
+---
+
+Laravel is awesome. Not just the framework, but its ecosystem. Let me walk you through it and show some of the beauties arround this community. 
+
+## Not just a fullstack framework
+First of all, I have to say that I don't participate as much as I want online, but I consume a lot from this community. In fact, I own a huge amount of skills I have to this community. The first thing that pops up on this topic is [Laracasts](http://laracasts.com). As the website says "it's like Netflix for developers". I should say that it's not only about Laravel, but PHP in general.
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: I think you meant "I owe" instead of "I own"?

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+There is also the **#laravel** IRC channel on freenode (which is pretty cool), [Laracasts Forum](https://laracasts.com/discuss), [LaravelIO](http://laravel.io/forum), [Larachat Slack Channel](http://larachat.co/), [Laravel News](https://laravel-news.com/) (a weekly newsletter about what is trending in the community), [Laravel Podcast](http://www.laravelpodcast.com/) (where thay talk about what is trending in the Laravel community), [Laravel Collective](http://laravelcollective.com/) (a group that maintains some packages that were extracted from the framework), [LaraJobs](https://larajobs.com/) (a place to find jobs, where I found [**madewithlove**](http://madewithlove.be/), haha), [Matt Stauffer's blog](https://mattstauffer.co/blog) is also a nice place to read about the trending Laravel features.
+
+I recommend a lot of other resources that I discovered on this community, like: [Full Stack Radio](http://fullstackradio.com/) (as the name says, a podcast aboute fullstack stuff, really good!), [Dev Duscussions](http://www.devdiscussions.com/), [Server for Hackers](https://serversforhackers.com/). I must have forgotten some pretty good resource, but I'll try to update this list.
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: Typo: Dev d**i**scussions

Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: Also: Sever**s** for hackers

Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: Also: resource**s**

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+## Development Environment
+Laravel provides some pretty good tools/features that speed up our development workflow, let's talk about some of them:
+
+### Homestead
+The first thing I should talk about is [Laravel's Homestead](http://laravel.com/docs/5.0/homestead). It's just a vagrant wrapper with a box that has most of the things a modern PHP application needs. Yes, you can still benefit from it even if you don't use Laravel. It has things like: Ubuntu 14.04, PHP 5.6, HHVM, nginx, MySQL, Postgres, Node (with Bower, Grunt, and Gulp), Redis, Memcached, Beanstalkd, Laravel Envoy, Blackfire Profiler. That being said, the whole point of homestead is that you don't need to build a VM for each project that you're working on, you can use the same VM for multiple projects, 'cause it's all there. Another great point is that everything is already baked inside the VM, so you don't have to wait it to provision.
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: you don't have to wait **for** it to provision.

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+However, I have to say that if your project differs from this setup, don't hesitate to build your own vagrant box with the tools you need.
+
+### Laravel (the framework)
+First things first, I should say that some people say things that I don't get it, like: "Laravel is too good, don't use it. It might not be arround in the next five years", what? I really don't get it. Laravel is an awesome framework. Use it wisely.
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: "that I don't get" (the last ''it" is unnecessary)

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+Laravel is a fullstack framework, it means that it provides a lot of funcionalioties, like: routing, database migrations, MVC structure (sort of, I'll talk about it in a second), authentication, billing service, command bus, pagination, validation, filesystem/cloud storage (same interface for both), an awesome ORM called Eloquent (which is an implementation of the ActiveRecord pattern).
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: funcionalioties -> I think just "functionality" is OK.

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+
+#### Asset management with Elixir
+As [Elixir's documentaion](http://laravel.com/docs/5.0/elixir) says:
+> Laravel Elixir provides a clean, fluent API for defining basic Gulp tasks for your Laravel application. Elixir supports several common CSS and JavaScript pre-processors, and even testing tools.
+
+It's a plugin for Gulp with a bunch of useful features like: compile less/sass, compile coffeescript, combine scripts/stylesheets, versioning your assets (so you don't need to worry about browser cache when you deploy front-end changes), browserify integration (so you can use EcmaScript 6 sintaxe right now).
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: Syntax, not syntaxe

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+#### Laravel's MVC structure
+One thing I would like to talk about is how Laravel changed the way I code. Reading Taylor's book [From Apprentice to Artisan](https://leanpub.com/laravel) he suggests get rid of the `app/models` folder. That was mind blowing to me. Like, your model is your domain and your domain is not a buch of database tables that interact. Your domain has behaviour. That is pretty good and one of the first benefits I saw when I decided to use this framework.
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: "Reading Taylor's book From Apprentice to Artisan he suggests get rid of..." Try to rewrite this sentence, it's a bit broken ;-)

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+If you come from a regular MVC framework where your have `app/controllers`, `app/models` and `app/views` folders, you might get a little confused on your first attempt to use Laravel (the version 5.x), but I assure you it will be better. It will feel better.
+
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: Read till here. Will review the rest later.

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+#### Background jobs
+Laravel ships with a *Queues* component that you can use to benefit from background jobs. For those who never used it before, think about a registration process with a welcome e-mail being sent at runtime. The user has to wait until the e-mail is sent correctly to actually use your app. Using *queues* you would just send the e-mail delivery to background (outside of the request scope), to be executed later or by another service. It would speed up the registration process and your user would still be happy with your welcome e-mail. In fact, he/she would be happier because you didn't make him/her wait.
+
+It shipts with a few queue drivers, like: redis, database, Amazon SQS, Beanstalkd, IronMQ and a "sync" that just executes the "job" at runtime. A good point is that every driver implements the same interface. It means that you can use any of these services to work with queues.
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: "It ships"

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+It's a plugin for Gulp with a bunch of useful features like: compile less/sass, compile coffeescript, combine scripts/stylesheets, versioning your assets (so you don't need to worry about browser cache when you deploy front-end changes), browserify integration (so you can use EcmaScript 6 sintaxe right now).
+
+#### Laravel's MVC structure
+One thing I would like to talk about is how Laravel changed the way I code. Reading Taylor's book [From Apprentice to Artisan](https://leanpub.com/laravel) he suggests get rid of the `app/models` folder. That was mind blowing to me. Like, your model is your domain and your domain is not a buch of database tables that interact. Your domain has behaviour. That is pretty good and one of the first benefits I saw when I decided to use this framework.
+
+If you come from a regular MVC framework where your have `app/controllers`, `app/models` and `app/views` folders, you might get a little confused on your first attempt to use Laravel (the version 5.x), but I assure you it will be better. It will feel better.
+
+#### Background jobs
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: This part is less readable than the others. I understood it, but I think if you don't know what queues are it's not so readable anymore...

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+It shipts with a few queue drivers, like: redis, database, Amazon SQS, Beanstalkd, IronMQ and a "sync" that just executes the "job" at runtime. A good point is that every driver implements the same interface. It means that you can use any of these services to work with queues.
+
+#### Broadcasting (WebSockets FTW!)
+This is still fresh at the moment I write this post: Laravel 5.1 is going to provide a websocket integration out of the box! It means that your can add realtime funcionalities to your app as easy as implementing an Interface on your backend code and a couple of JS lines on your front-end code. How cool is that? You can see an example at [laracasts](https://laracasts.com/lessons/broadcasting-events-in-laravel-5-1) where taylor presents this feature.
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: Taylor with a capital 'T'

Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: funcionalities -> func**t**ionalities

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+
+There are a lot of good stuff in the framework that I would like talk about, but this post would get huge. Instead, you can walk arround the [documentation](http://laravel.com/docs/5.0/) and there is a lot of good stuff on [Laracasts](http://laracasts.com).
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: There are a lot of good stuff -> there is a lot of good stuff

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+
+#### Lumen, Laravel's little (and faster) brother
+So you found a context in your application that would be faster if didn't have to build the whole framework to do the job? You can use [Lumen](http://lumen.laravel.com/) which is the official Laravel micro-framework. It is faster than Laravel (and faster than other micro-frameworks, as the main page says), but you still use some of Laravel's cool features, such as the Eloquent ORM and the Queues Component.
+
+It's also possible to use some other Laravel's fullstach funcionalities, but if you need some of it you might consider moving to the fullstack framework.
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: fullstach -> fullstack

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+
+Let's now talk about some of the goodies Laravel provides for your deployment.
+
+## Deploy
+There were times where deploying an application was a pain, with lots of FTP files being sent and failures and so on. One of the first things we have to do to deploy our application is build a server with all the funcionalities we need. Laravel has a service for that called [Forge](https://forge.laravel.com/).
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: funcionalities -> functionalities

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+
+### Laravel Forge 
+It handles the creation and provisioning of servers for you. It basically hooks into your cloud provider (such as Amazon, DigitalOcean or Linode) and create your servers so you don't have to worry about "which version of PHP is there?", "is mysql installed?" or things like that. Pretty cool. You can even automate your deploys there using git, but Laravel also provides a service for deployments called [Envoyer](https://envoyer.io/).
+
+### Laravel Envoyer
+While you can have automatic deployment setup using Forge, it's recommended that you use a dedicated service for that. That is what Envoyer is all about. It makes deploy new code a breeze. It's zero downtime, so your users won't face a "maintenance" page while you deploy new stuff.
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: it makes deploy**ing** new code?

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+
+It also have really cool stuff like cron job monitoring (which is useful, so you get notifications if one of your application's cron jobs stops working) and application health check. Deploying PHP applications were never as easy as now.
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: It also have -> it also has
Deploying PHP applications were never -> was never

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+
+### Rocketeer
+[Rocketeer](http://rocketeer.autopergamene.eu/#/docs/rocketeer/README) is a tool for automating deploys. It seats side by side with ruby's Capistrano or python's Fabric. With that you can easily setup your own deploy process with zero downtime, notifications and so on. Pretty cool!
+
+### Envoy
+Laravel also provides another tool for task running called [Envoy](http://laravel.com/docs/5.0/envoy). With this tool, you can easily run a few tasks on your remote servers or also setup a simple deploy script, as you can see on this [Server for Hackers](https://serversforhackers.com/video/deploying-with-envoy-cast) screencast.
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: Server**s** for hackers

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+
+## Conclusion
+These are just a few goodies that Laravel provides to us. Once you start working with laravel, you notice how good it is and how easy it is to use it's features/services.
+
+Last but not least, becareful with your application, the framework is just a tool, but you can follow SOLID principles and still benefit from Laravel's goodies. Take a look into the [Contracts](http://laravel.com/docs/5.0/contracts), which is a component used in the framework so you can easily depend on Interfaces to use the framework features and [Laravel's IoC](http://laravel.com/docs/5.0/container) will handle the instantiation of the right objects.
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: becareful -> be careful

File: _posts/2015-05-08-laravel-ecosystem.md [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+It also have really cool stuff like cron job monitoring (which is useful, so you get notifications if one of your application's cron jobs stops working) and application health check. Deploying PHP applications were never as easy as now.
+
+### Rocketeer
+[Rocketeer](http://rocketeer.autopergamene.eu/#/docs/rocketeer/README) is a tool for automating deploys. It seats side by side with ruby's Capistrano or python's Fabric. With that you can easily setup your own deploy process with zero downtime, notifications and so on. Pretty cool!
+
+### Envoy
+Laravel also provides another tool for task running called [Envoy](http://laravel.com/docs/5.0/envoy). With this tool, you can easily run a few tasks on your remote servers or also setup a simple deploy script, as you can see on this [Server for Hackers](https://serversforhackers.com/video/deploying-with-envoy-cast) screencast.
+
+## Conclusion
+These are just a few goodies that Laravel provides to us. Once you start working with laravel, you notice how good it is and how easy it is to use it's features/services.
```
Comment (hannesvdvreken) [inline _posts/2015-05-08-laravel-ecosystem.md]: it's -> its

**Comment by hannesvdvreken** [comment]
Nice post!

Maybe re-read the conclusion part and the queues part a few times. I think those two can improve.

Consider just mentioning Socialite and Cachier for completeness.

I'd love to see this published. I know Dries Vints will be more than happy to read and share it.

File: _posts/2015-05-08-laravel-ecosystem.md (hunk 13 above)
Comment (bramdevries) [inline _posts/2015-05-08-laravel-ecosystem.md]: > So you found a context in your application that would be faster if didn't have to build the whole framework to do the job?

This reads weird maybe "So you found a context in your application that would be faster if it didn't need to use the whole framework"

File: _posts/2015-05-08-laravel-ecosystem.md (hunk 14 above)
Comment (bramdevries) [inline _posts/2015-05-08-laravel-ecosystem.md]: There were times when

**Comment by tonysm** [comment]
@hannesvdvreken @bramdevries thanks, guys! Great feedback, lots of typos and grammar fixes.. Hannes, I'll add Cachier and Socialite sections to this post, they definitely deserve to be mentioned. I'll also rewrite the queues and conclusion sections. Thanks!

**Comment by dieterve** [comment]
I like it already, nice one.

**Comment by tonysm** [comment]
@hannesvdvreken I added the Cachier and Socialite sections and tried to improve the Queues and Conclusion sections, what do you think now? If you have any suggestion for those sections, let me know

**Comment by hannesvdvreken** [comment]
:+1: almost ready

**Comment by hannesvdvreken** [comment]
Hit that green button when you're ready
screen shot 2015-05-14 at 22 29 18
. :ship:

**Comment by tonysm** [comment]
For a second I thought I had zoomed in the page..

**Pull request merged**